### ISSUE TITLE
Performance of TFN on MOSI dataset

### ISSUE THREAD

**Issue by Jongchan** [A2Zadeh/CMU-MultimodalDataSDK#25]
First of all, thank you for awesome dataset, models and this convenient interface.

For baseline/test purpose, I've ran TFN on MOSI dataset without changing any code.
```
python train.py --epochs 100 --patience 10
```

After training, the final log for validation/test accuracy is as below
```
Validation loss is: 1.11342244377
Validation binary accuracy is: 0.672489082969
MAE on test set is 1.14087029727
Binary accuracy on test set is 0.673469387755
Precision on test set is 0.620289855072
Recall on test set is 0.697068403909
F1 score on test set is 0.656441717791
Seven-class accuracy on test set is 0.304664723032
Correlation w.r.t human evaluation on test set is 0.502070320118
```

The network seems to be trained for 28 epochs. The full log is [here](https://drive.google.com/file/d/17QS58sfNDG5geFhwQJwndfL3kXeWnBtt/view?usp=sharing).

The performance is quite different from the one in TFN paper.
The validation performance seems to fluctuate along training, so the best checkpoint should be set somewhere before epoch 28.
Is this an expected output? or is there any detail missing in the baseline code?

**Comment by Justin1904** [comment]
Hi @Jongchan, there are some possible reasons why you didn't get the same numbers as the paper:

1. the default hyper parameter settings in the TFN repo is not the optimal one, in fact it is just a randomly selected setting.

2. the original paper used feature selection on input visual and acoustic data. As feature selection is not actually part of the model, we didn't include that part in this implementation. This seems to interact with the choice of optimal hyper parameters as well.

3. the original paper evaluated the model based on a different data split than the current one, which affect the evaluation.

In all, in our previous experiments, it is possible to get around 73-75 binary accuracy for some hyper parameter settings with this implementation, and that would be considered normal (and close to the latest sota as well). Generally if you see a validation loss above 1.10, it is very likely that the model just haven't converged yet, and you can probably evade that problem by increasing the "--patience" argument for training, as well as doing some grid search on hyper parameters.

**Comment by Jongchan** [comment]
@A2Zadeh @Justin1904 
Thank you for your kind reply. I thought currently included TFN is the final version. 
I will review the PyTorch version with the original implementation. If some other things come up, I raise another issue. Thanks!

**Comment by Lhemamou** [comment]
Hi @Justin1904 thanks for your reply :) Could you explain better the feature selection used in order to reproduce the results? Thanks !
